### ISSUE TITLE
SkiaSharp, SVG & Windows

### ISSUE THREAD

**Issue by PI2R** [xamarin/MobileBlazorBindings#274]
Hi,
I have a project targeting macOS, windows, and eventually iOS & Android.
I needed to display some svg icons and was looking for a possible solution.
Therefore I ended up trying to use SkiaSharp.
I managed to get a basic drawing works on Android but when I run the windows project I get an exception stating that it cannot find the assembly SkiaSharp.Views.Forms even though I added it through nuget.
I started to gather clues that it might not work on Windows applications, but reading through it I found this article from Matthew saying:

> "SkiaSharp can be used to render images on many platforms, including iOS, macOS, tvOS, Android, and Universal Windows Platform (UWP). Not only does SkiaSharp support native mobile apps, it also supports .NET Standard 1.3 and .NET Core on macOS and Windows, as well as Classic Windows Desktop and WPF applications."
(https://devblogs.microsoft.com/xamarin/drawing-with-skiasharp/)

This is an article dated back from 2017 for Xamarin, but I don't see why it wouldn't apply here as this is supposed to be just a wrapper around it.

My questions:
1) What is it? Should I continue trying to make it work or is this pointless?
2) Is there any other alternative to display svg images on all platforms?

**Comment by Eilon** [comment]
I don't think I've tried SkiaSharp in WPF so I'll try to see what's going on.

Tagging @lachlanwgordon who did the original SkiaSharp work in case he has any idea.

**Comment by lachlanwgordon** [comment]
Hi @PI2R, this should work but I haven't tested it so I'll take a look.

Are you using UWP or WPF?

**Comment by lachlanwgordon** [comment]
If it's WPF I think I've found the issue, you also need to add the package `SkiaSharp.Views.Forms.WPF` to your WPF project, similar to how you need to have `Xamarin.Forms.Platform.WPF` in the WPF project even though you don't need the equivalent on other platforms.

**Comment by PI2R** [comment]
Hi @lachlanwgordon 
It is the default one created by the template, so unless I am going crazy (or crazier) this should be WPF :-)
Looking at the issue I already found on some Xamarin issues the fact that a special WPF package was need for SkiaSharp.
Unfortunately I installed SkiaSharp.Views..WPF instead of SkiaSharp.Views.Forms.WPF...
I tried to install the one you suggested and indeed it worked! Thanks a lot

**Comment by Eilon** [comment]
I've updated the docs on this to add a note for WPF projects: https://github.com/MicrosoftDocs/mobile-blazor-bindings/pull/33. It should go live at https://docs.microsoft.com/mobile-blazor-bindings/walkthroughs/skiasharp shortly.

@PI2R  thank you for reporting this!
@lachlanwgordon thank you for assisting!

**Comment by PI2R** [comment]
Hi @Eilon 
Thank you for updating the docs.
There was a test I wanted to do and that I just did when I saw you updated the doc.

In fact it seems I was half right and that SkiaSharp.Views.WPF is also required. 
Indeed when @lachlanwgordon suggested to add the nuget package I did it without removing the one I added before.
When I just tried to remove it it stopped working again.

**Comment by Eilon** [comment]
Ah interesting I will update the docs to reflect that, thank you for the info!

**Comment by PI2R** [comment]
Also while you're at it, I just tried on macOS, and adding package SkiaSharp.Views.Forms allowed me to draw the same canvas.
You might want to add macOS to your bullet iOS and Android.

**Comment by Eilon** [comment]
Thank you once again @PI2R , the docs are updated now: https://github.com/MicrosoftDocs/mobile-blazor-bindings/pull/34 (will be live soon)

**Comment by PI2R** [comment]
I saw you added WIndows as well to the list of project types that should have SkiaSharp.Views.Forms installed.
It does not hurt and I don't think it is worth you waste time on this, but FYI my tests worked without it. Only the 2 "SkiaSharp.*.WPF" packages were needed.
Just to be sure I installed this extra package to the windows project and it did not seem to cause any more troubles to have it.
And maybe you're right it could be needed for more complex SkiaSharp scenarios.
